slam-libraries/viam-cartographer/src/slam_service/slam_service_test.cc

```diff
@@ -69,45 +72,308 @@ BOOST_AUTO_TEST_CASE(OverwriteMapBuilderParameters_set_values_mapping) {
+    boost::filesystem::path tmpdir =
+        utils::createTmpDirectoryAndAddFiles(data_files, map_files);
+    slamService.path_to_map = tmpdir.string() + "/map";
+    slamService.DetermineActionMode();
+
+    BOOST_TEST(slamService.GetActionMode() == SLAMServiceActionMode::UPDATING);
+    slamService.OverwriteMapBuilderParameters();
+    checkCartoMapBuilderParameters(slamService);
+
+    // Remove the temporary directory and its contents
+    utils::removeTmpDirectory(tmpdir);
+}
+
+BOOST_AUTO_TEST_CASE(OverwriteMapBuilderParameters_set_values_localizing) {
+    SLAMServiceImpl slamService;
+    slamService.optimize_every_n_nodes = 9999;
+    slamService.num_range_data = 9998;
+    slamService.missing_data_ray_length = 9997.77;
+    slamService.max_range = 9996.66;
+    slamService.min_range = 9995.55;
+    slamService.max_submaps_to_keep = 9994;
+    slamService.fresh_submaps_count = 9993;
+    slamService.min_covered_area = 9992.22;
+    slamService.min_added_submaps_count = 9991;
+    slamService.occupied_space_weight = 9990.09;
+    slamService.translation_weight = 9989.89;
+    slamService.rotation_weight = 9988.88;
+
+    // Set the action mode to localizing by providing an apriori map
+    // and by setting map_rate_sec == 0
+    slamService.map_rate_sec = std::chrono::seconds(0);
+
+    // Create a temp directory with a map file in it
+    std::vector<std::string> data_files{};
+    std::vector<std::string> map_files{
+        "map_data_2022-02-11T01:44:53.1903Z.pbstream"};
+    // Create a unique path in the temp directory and add the files
+    boost::filesystem::path tmpdir =
+        utils::createTmpDirectoryAndAddFiles(data_files, map_files);
+    slamService.path_to_map = tmpdir.string() + "/map";
+    slamService.DetermineActionMode();
+
+    BOOST_TEST(slamService.GetActionMode() ==
+               SLAMServiceActionMode::LOCALIZING);
     slamService.OverwriteMapBuilderParameters();
     checkCartoMapBuilderParameters(slamService);
+
+    // Remove the temporary directory and its contents
+    utils::removeTmpDirectory(tmpdir);
 }
 
-BOOST_AUTO_TEST_CASE(GetActionMode_mapping) {
+BOOST_AUTO_TEST_CASE(
+    OverwriteMapBuilderParameters_check_default_values_localizing) {
     SLAMServiceImpl slamService;
+
+    // Set the action mode to localizing by providing an apriori map
+    // and by setting map_rate_sec == 0
+    slamService.map_rate_sec = std::chrono::seconds(0);
+
+    // Create a temp directory with a map file in it
+    std::vector<std::string> data_files{};
+    std::vector<std::string> map_files{
+        "map_data_2022-02-11T01:44:53.1903Z.pbstream"};
+    // Create a unique path in the temp directory and add the files
+    boost::filesystem::path tmpdir =
+        utils::createTmpDirectoryAndAddFiles(data_files, map_files);
+    slamService.path_to_map = tmpdir.string() + "/map";
+    slamService.DetermineActionMode();
+
+    BOOST_TEST(slamService.GetActionMode() ==
```

Comment:
I don't understand. Can you give an example of what you're thinking about and the motivation for doing it?

Reply:
`checkCartoMapBuilderParameters` contains `MapBuilder` because it checks only the `map_builder` related parameters, which are only the internal cartographer (hyper) parameters and none of the other `slamService` specific parameters.